crates/accelerate/src/gate_direction.rs

```diff
@@ -142,8 +159,479 @@ where
+                            "The circuit requires a connection between physical qubits {:?} for {}",
+                            op_args,
+                            packed_inst.op.name()
+                        )));
+                    }
+                }
+                _ => {}
+            }
+        }
+        // No matching replacement found
+        if gate_complies(packed_inst, &[op_args1, op_args0])
+            || has_calibration_for_op_node(py, dag, packed_inst, &[op_args1, op_args0])?
+        {
+            return Err(TranspilerError::new_err(format!("{} would be supported on {:?} if the direction was swapped, but no rules are known to do that. {:?} can be automatically flipped.", packed_inst.op.name(), op_args, vec!["cx", "cz", "ecr", "swap", "rzx", "rxx", "ryy", "rzz"])));
+            // NOTE: Make sure to update the list of the supported gates if adding more replacements
+        } else {
+            return Err(TranspilerError::new_err(format!(
+                "{} with parameters {:?} is not supported on qubits {:?} in either direction.",
+                packed_inst.op.name(),
+                packed_inst.params_view(),
+                op_args
+            )));
+        }
+    }
+
+    for (node, op_blocks) in ops_to_replace {
+        let packed_inst = dag.dag()[node].unwrap_operation();
+        let OperationRef::Instruction(py_inst) = packed_inst.op.view() else {
+            panic!("PyInstruction is expected");
+        };
+        let new_op = py_inst
+            .instruction
+            .bind(py)
+            .call_method1("replace_blocks", (op_blocks,))?;
+
+        dag.py_substitute_node(dag.get_node(py, node)?.bind(py), &new_op, false, false)?;
+    }
+
+    for (node, replacemanet_dag) in nodes_to_replace {
+        dag.py_substitute_node_with_dag(
+            py,
+            dag.get_node(py, node)?.bind(py),
+            &replacemanet_dag,
+            None,
+            true,
+        )?;
+    }
+
+    Ok(dag)
+}
+
+// Check whether the dag as calibration for a DAGOpNode
+fn has_calibration_for_op_node(
+    py: Python,
+    dag: &DAGCircuit,
+    packed_inst: &PackedInstruction,
+    qargs: &[Qubit],
+) -> PyResult<bool> {
+    let py_args = PyTuple::new_bound(py, dag.qubits().map_indices(qargs));
+
+    let dag_op_node = Py::new(
+        py,
+        (
+            DAGOpNode {
+                instruction: CircuitInstruction {
+                    operation: packed_inst.op.clone(),
+                    qubits: py_args.unbind(),
+                    clbits: PyTuple::empty_bound(py).unbind(),
+                    params: packed_inst.params_view().iter().cloned().collect(),
+                    extra_attrs: packed_inst.extra_attrs.clone(),
+                    #[cfg(feature = "cache_pygates")]
+                    py_op: packed_inst.py_op.clone(),
+                },
+                sort_key: "".into_py(py),
+            },
+            DAGNode { node: None },
+        ),
+    )?;
```

Comment:
This is something I discussed with @kevinhartman, but we will probably move to using `.into_py(py)` to convert an `PyClass` to a `PyObject` in Rust, but we want to make sure to do it in a follow up to make sure it is used everywhere, so I'm okay with leaving it as it is here but perform the change to `.into_py(py)` in a follow up.